src/payment/strategies/adyenv2/adyenv2-payment-strategy.spec.ts

```diff
@@ -242,318 +116,337 @@ describe('AdyenV2PaymentStrategy', () => {
-                        },
-                    });
+                try {
+                    await strategy.execute(getOrderRequestBody());
+                } catch (error) {
+                    expect(error).toBeInstanceOf(RequestError);
+                }
 
-                    return;
-                }),
-                unmount: jest.fn(),
-            };
+                expect(paymentActionCreator.submitPayment).toHaveBeenCalledTimes(1);
+                expect(adyenCheckout.create).toHaveBeenCalledTimes(2);
+            });
 
-            jest.spyOn(adyenCheckout, 'create')
-                .mockImplementationOnce(jest.fn((_type, options) => {
-                    componentOptions = options;
+            it('tries to pay with invalid component state', async () => {
+                const adyenInvalidPaymentComponent = {
+                    mount: jest.fn(),
+                    unmount: jest.fn(),
+                };
+                jest.spyOn(adyenCheckout, 'create')
+                    .mockReturnValue(adyenInvalidPaymentComponent);
 
-                    return threeDS2ChallengeComponent;
-                }));
+                await strategy.initialize(options);
 
-            jest.spyOn(adyenV2ScriptLoader, 'load').mockReturnValue(Promise.resolve(adyenCheckout));
+                try {
+                    await strategy.execute(getOrderRequestBody());
+                } catch (error) {
+                    expect(error).toBeInstanceOf(NotInitializedError);
+                }
+
+                expect(paymentActionCreator.submitPayment).toHaveBeenCalledTimes(0);
+                expect(adyenCheckout.create).toHaveBeenCalledTimes(2);
+            });
 
-            jest.spyOn(paymentActionCreator, 'submitPayment')
-                .mockReturnValueOnce(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, identifyShopperError)))
-                .mockReturnValueOnce(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, challengeShopperError)))
-                .mockReturnValueOnce(submitPaymentAction);
-
-            await strategy.initialize(options);
-            await strategy.execute(getOrderRequestBody());
-
-            expect(paymentActionCreator.submitPayment).toHaveBeenCalledTimes(3);
-            expect(paymentActionCreator.submitPayment).toHaveBeenLastCalledWith({
-                methodId: 'scheme',
-                paymentData: {
-                    nonce: JSON.stringify({
-                        threeDS2Token: 'challengeToken',
-                        paymentData: 'paymentData',
+            it('pays with vaulted instrument', async () => {
+                jest.spyOn(paymentActionCreator, 'submitPayment')
+                    .mockReturnValueOnce(submitPaymentAction);
+
+                await strategy.initialize(options);
+                await strategy.execute(getOrderRequestBodyWithVaultedInstrument());
+
+                expect(paymentActionCreator.submitPayment).toHaveBeenCalledTimes(1);
```

Comment:
not required, but it's recommended to add some expectations around the arguments
e.g.
```hs
expect(paymentActionCreator.submitPayment).toHaveBeenCalledWith(expect.objectMatching({
  paymentData: ....,
}));
```